libbeat/beat/event.go

```diff
@@ -110,45 +110,50 @@ func (e *Event) deepUpdate(d mapstr.M, overwrite bool) {
 	if len(d) == 0 {
 		return
 	}
-	fieldsUpdate := d.Clone() // so we can delete redundant keys
 
-	var metaUpdate mapstr.M
+	// TODO: Refactor to dynamic field removal/assignment when needed
+	timestampValue, timestampExists := d[timestampFieldKey]
```

Comment:
```suggestion
	// it's supported to update the timestamp using this function,
	// however, we must handle it separately, since it's a separate field of the event
	timestampValue, timestampExists := d[timestampFieldKey]
```